### PR TITLE
Fix compatibility issue with ST4114+

### LINKED ISSUE #213: ST4114 incompatibility
ST's HTML syntax had to change the way how CSS/JS syntax definitions are embedded within style and script tags in order to fix a core bug. It also requires changes to how those syntaxes are to be embedded. This change unfortunatelly effects all syntaxes, which inherit HTML.

Related PR: https://github.com/sublimehq/Packages/pull/2831

This broke VUE syntax.

A short look at this package revealed the following lines to be obsolete or not needed at all.

https://github.com/vuejs/vue-syntax-highlight/blob/6c3c26d6a022b2b4353e5a7f98c5e1496e9f7567/Vue%20Component.sublime-syntax#L191-L195

A backward compatible fix was to remove the context.

An inherited syntax needs to override `script-javascript-content` or `style-css-content` only, if it wants to embed a custom CSS/JS syntax (or scope), which is not the case here.

--- 

I'd suggest to take a look at...

https://github.com/sublimehq/Packages/blob/c70117be6402c54ccf860ab8edfcba8aade46c9e/ASP/HTML%20(ASP).sublime-syntax#L37-L67

This approach might be needed for all kinds of embedded script syntaxes to ensure toggle comment is working on the first and last line properly.

### PR DESCRIPTION
**Pull request by deathaxe**
Fixes #213

Superseeds #214

This PR applies the required patch of ST4114's HTML.sublime-syntax in order to fix VUE's incompatibility with most recent ST, while maintaining backward compatibility with older builds.

The relevant contexts are overridden and variables are copied over from HTML.sublime-syntax.

Note: 

1. The `style-close-tag` was probably not needed, as it hasn't changed, but it is copied to maintain formal consistency with script tags.
2. This PR exists to as help was asked for in #214.

### CODE REVIEW

**Comment by skyronic** [comment]
Hey @deathaxe @rchl thank you for taking care of this! 

I'd like to add some tests for the ST4 branch, I'd started on some early tests for #210 but do you think there are any tests we can add?

Let me know if you think this looks good, I can go ahead and make a release where ST4 gets the new branch.

I'll test this out myself. To be honest - I haven't had time to look at this in detail - but I'll definitely spend some time on this today.

**Comment by rchl** [comment]
What I can provide is in-the-field testing since I work with Vue files daily. Have been running with this change since yesterday and haven't noticed any issues.

@skyronic isn't there some blocking issue that makes some testing impossible?

@deathaxe can you provide a small example showing the superiority of this change vs. #214? I would just like to know what it fixes specifically that the other one didn't and test more stuff around that.

**Comment by deathaxe** [comment]
Added some tests and a Github Action to run them automatically. They should become active once, this PR is merged.

I took the config from https://github.com/SublimeText-Markdown/MarkdownEditing/blob/3.0.0/.github/workflows/ci-syntax-tests.yml

**Comment by deathaxe** [comment]
Already found various other improvements, but I'll hold them off until this one is merged.

**Comment by skyronic** [comment]
This is some absolutely stellar stuff, @deathaxe. Thank you!!

I tested it locally and it works great! Let me know when you're happy with it, and I'll merge and release.

**Comment by skyronic** [comment]
> @skyronic isn't there some blocking issue that makes some testing impossible?

@rchl - yes, there was indeed an issue I encountered while testing it. Something like 

```
<style lang="sass">
@import '~foo'
// ^ source.sass
a
  color: red
</style>
```

Would not work, because even though the `//` would be recognized as a comment, Syntax Tests needed to be wrapped inside a `<!-- -->`

I'm still not 100% sure how @deathaxe's tests work but it appears to be using HTML comment start/end punctuation in `<script>` and `<style>` tags. I would need to take a closer look.

**Comment by deathaxe** [comment]
> Would not work, because even though the // would be recognized as a comment, Syntax Tests needed to be wrapped inside a `<!-- -->`

The comment character used for syntax tests is defined by the first line of the test file. Your test branch uses those html comments in the first line, which would mean to wrap all test cases in such. That's more than cumbersome. Hence core HTML.sublime-syntax uses `##`. I've just choosen `//` because it seems more common.

**Comment by deathaxe** [comment]
I found some bug in template tag handling, I am not sure about the way `mustache-expression` is implemented and would suggest to reorganize the syntax a bit to group contexts which belong together and sort those groups to match the order of core HTML.sublime-syntax.

All of that could happen soon, but feels out of scope for this PR.

I'd open new ones for the mentioned fixes.

File: .github/workflows/ci-syntax-tests.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    branches:
+      - 'master'
+    paths:
+      - '.github/workflows/ci-syntax-tests.yml'
+      - '**.sublime-syntax'
+      - '**/syntax_test_*'
+      - '**.tmPreferences'
+  pull_request:
+    branches:
+      - '**'
```
Comment (rchl) [inline .github/workflows/ci-syntax-tests.yml]: Should be targeting the `st4` branch, not `master`.

**Review by rchl — COMMENTED** [review]
Again, I can't provide a proper review but looks good when testing the changes locally.

File: .github/workflows/ci-syntax-tests.yml (hunk 1 above)
Comment (deathaxe) [reply .github/workflows/ci-syntax-tests.yml]: st4 is no working branch, which is merged back to master?

Comment (deathaxe) [reply .github/workflows/ci-syntax-tests.yml]: I mean, I'd add an st3 branch to lock the st3 state or do some st3 specific fixes and go on with master.

Comment (rchl) [reply .github/workflows/ci-syntax-tests.yml]: But currently it's set up in a way where ST4 releases are made from the `st4` branch.

I believe the reason why it's not the other way around is that the main/master branch contains also non-ST-specific syntaxes so it wouldn't fully make sense to make the ST4-specific branch the main one.

Screenshot 2021-09-08 at 21 31 18

File: .github/workflows/ci-syntax-tests.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    branches:
+      - 'st4'
+    paths:
+      - '.github/workflows/ci-syntax-tests.yml'
+      - '**.sublime-syntax'
+      - '**/syntax_test_*'
+      - '**.tmPreferences'
+  pull_request:
+    branches:
+      - '**'
```
Comment (rchl) [inline .github/workflows/ci-syntax-tests.yml]: This probably should be `st4` too? We only want to check PRs that are made against the ST4 branch.

That said, I wonder if the CI configuration actually has to live on the main/master branch regardless...

File: .github/workflows/ci-syntax-tests.yml (hunk 1 above)
Comment (skyronic) [reply .github/workflows/ci-syntax-tests.yml]: We don't need tests for ST3 version.

The `st4` branch will be the working branch for all ST4-compatible releases. I don't plan to update anything for the ST3 version of the plugin unless any significant issues crop up. You can safely target tests only for the new version and the `st4` branch.

File: .github/workflows/ci-syntax-tests.yml (hunk 2 above)
Comment (skyronic) [reply .github/workflows/ci-syntax-tests.yml]: I double-checked by merging this PR in my fork with an `st4` branch, and the GH actions seem to run fine!

https://github.com/skyronic/vue-syntax-highlight/tree/st4

The tests are failing but that's not specific to GH actions.

Comment (deathaxe) [reply .github/workflows/ci-syntax-tests.yml]: I've choosen the wrong package name.

File: .github/workflows/ci-syntax-tests.yml (hunk 1 above)
Comment (deathaxe) [reply .github/workflows/ci-syntax-tests.yml]: OK, the commit below targets push's to st4 only.

File: .github/workflows/ci-syntax-tests.yml (hunk 2 above)
Comment (deathaxe) [reply .github/workflows/ci-syntax-tests.yml]: If someone creates a PR for a PR, syntax tests should probably run, too. So targeting a dedicated target branch is probably not a good idea. Not sure how likely it is in this repo, but I've done such a couple of times.

**Comment by deathaxe** [comment]
I've removed Github Actions for now as it appears to be more work than expected.

It probably fails due to missing embedded syntax packages. Investigating and fixing it is really out of scope here.

**Comment by deathaxe** [comment]
This PR contains all fixes required to run with all ST4 releases. I wouldn't want to add more stuff in here.

**Comment by skyronic** [comment]
Looks good to me, tested it and works great! Will submit to package control today!

Thanks again, @deathaxe @rchl

**Pull request merged**